### ISSUE TITLE
Proposed Analysis: WXS cnv annotated seg file generation

### ISSUE THREAD

**Issue by kgaonkar6** [PediatricOpenTargets/ticket-tracker#102]
#### What analysis module should be updated and why?
Add gene overlap annotation to cnvkit seg file for WXS samples

#### What changes need to be made? Please provide enough detail for another participant to make the update.
Run the code
https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d53b626c4eae5dccc4efdfda0432253b53338713/analyses/focal-cn-file-preparation/run-prepare-cn-OpenTarget.sh#L42-L80

#### What input data should be used? Which data were used in the version being updated?
v7
cnv-cnvkit.seg.gz

#### When do you expect the revised analysis will be completed?
1-2 days

**Comment by jharenza** [comment]
@runjin326 can you pick this up early next week for preparation for data release #92 ?

**Comment by jharenza** [comment]
The file is staged at https://kf-openaccess-us-east-1-prd-pbta.s3.amazonaws.com/open-targets/v7/cnv-cnvkit.seg.gz right now

**Comment by runjin326** [comment]
@jharenza, yes! I will work on this and get it done by early next week :)

**Comment by runjin326** [comment]
Just to clarify a couple details:
1) We only use v7 `cnv-cnvkit.seg.gz` (and the output from running 01, `cnvkit_with_status.tsv)` to run the script, right? Other input such as `cnv-consensus.seg.gz` will still be what we currently have in the `data` folder? 
2) Some lines such as ones [calling expression data and independent samples](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d53b626c4eae5dccc4efdfda0432253b53338713/analyses/focal-cn-file-preparation/run-prepare-cn-OpenTarget.sh#L71-L72) and the [one defining tumors](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d53b626c4eae5dccc4efdfda0432253b53338713/analyses/focal-cn-file-preparation/rna-expression-validation.R#L119) are out of date. Do we modify the script as well?
3) The independent samples in the scripts uses the DNA one - should we use the RNA one instead? 
4) Final deliverables will be a PR right?

**Comment by kgaonkar6** [comment]
> Just to clarify a couple details:
> 
> 1. We only use v7 `cnv-cnvkit.seg.gz` (and the output from running 01, `cnvkit_with_status.tsv)` to run the script, right? Other input such as `cnv-consensus.seg.gz` will still be what we currently have in the `data` folder?

Yes you will need v7 cnv-cnvkit.seg.gz` and `histologies.tsv` and then run, additional info [here](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/tree/dev/analyses/focal-cn-file-preparation#running-this-analysis) :
`RUN_ORIGINAL=1 bash analyses/focal-cn-file-preparation/run-prepare-cn.sh`

> 2. Some lines such as ones [calling expression data and independent samples](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d53b626c4eae5dccc4efdfda0432253b53338713/analyses/focal-cn-file-preparation/run-prepare-cn-OpenTarget.sh#L71-L72) and the [one defining tumors](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/d53b626c4eae5dccc4efdfda0432253b53338713/analyses/focal-cn-file-preparation/rna-expression-validation.R#L119) are out of date. Do we modify the script as well?
> 3. The independent samples in the scripts uses the DNA one - should we use the RNA one instead?
Yes that's right! We will need to update the independent to RNA ones, and also update the `composition` filtering to `composition == "Solid Tissue" | composition == "Bone Marrow"`

> 4. Final deliverables will be a PR right?
I think this will be a muli-step analysis
-  PR to update to use the independent RNA files as input for `rna-expression-validation.R` 
-  PR to update composition filter `composition == "Solid Tissue" | composition == "Bone Marrow"` and rerun

**Comment by runjin326** [comment]
@kgaonkar6, so I made the modifications but it kept getting killed at the step that [does the annotation](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/8bc000c83cc9125020f66515427874bd0b112719/analyses/focal-cn-file-preparation/04-prepare-cn-file.R#L241-L248) Any idea how I can fix that? My docker is already set to maximum for all resources - but I don't VPN for EC2 instance yet.

**Comment by kgaonkar6** [comment]
I think a input param to filter for WXS in [04-prepare-cn-file.R](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/dev/analyses/focal-cn-file-preparation/04-prepare-cn-file.R) might help with your memory issue because of  reduced number of samples . Currently `cnv-cnvkit.seg.gz`has WGS and WXS

**Comment by runjin326** [comment]
@kgaonkar6 , so it would be:
`WES_samples <- histologies_df %>% filter(experimental_strategy == "WXS") %>% pull(Kids_First_Biospecimen_ID) %>% unique()` 
`cnv_df_subset <- cnv_df %>% filter(Kids_First_Biospecimen_ID %in% WES_samples)` 
And run the annotation using `cnv_df_subset` correct?

**Comment by kgaonkar6** [comment]
correct, just wanted to mention that we will have this as an input param `--runWXSonly==TRUE` to filter of WXS only

**Comment by runjin326** [comment]
Good point! I almost went ahead to change the code to just do the WES lol

**Comment by runjin326** [comment]
I actually have a few more questions. In [this step](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/8bc000c83cc9125020f66515427874bd0b112719/analyses/focal-cn-file-preparation/rna-expression-validation.R#L131-L136), it says we are only looking at non tumor samples. However, why do we have to add `composition` as a filter? Also, now that we have independent RNA, could we just deleted [these lines](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/8bc000c83cc9125020f66515427874bd0b112719/analyses/focal-cn-file-preparation/rna-expression-validation.R#L116-L146)? 

And for selecting WXS, I kept confusing myself. Should we  actually do the following: `Kids_First_Participant_ID` with WXS specimens, select the `Kids_First_Biopecimen_ID` of the "RNA-seq" samples from those participants and use those `Kids_First_Biopecimen_ID` to subset the CNV files, right? If we use `Kids_First_Biopecimen_ID` for WXS to subset, this will give us DNA samples and it would not have any overlap with RNA independent samples ([called in this step](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/8bc000c83cc9125020f66515427874bd0b112719/analyses/focal-cn-file-preparation/rna-expression-validation.R#L158-L167)), right? 

Lastly, for the [z-score step](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/8bc000c83cc9125020f66515427874bd0b112719/analyses/focal-cn-file-preparation/util/rna-expression-functions.R#L25-L31), could we subset the expression file to the individual samples first? This step also keep crashing on me.
